src/twfy_tools/utils/contact_io.py

```diff
@@ -0,0 +1,91 @@
+"""
+Interface to upload user optin information to MailChimp.
+"""
+
+from datetime import date, timedelta
+
+from typer import Typer
+
+from twfy_tools.common.config import config
+from twfy_tools.common.mailchimp import (
+    InterestInternalId,
+    MailChimpHandler,
+    MemberAndInterests,
+)
+from twfy_tools.db.models import OptinValues, User
+
+mailing_list_name = "mySociety Newsletters"
+
+app = Typer()
+
+
+def get_internal_optin_id(
+    interest_group: str, interest_name: str
+) -> InterestInternalId:
+    client = MailChimpHandler(config.MAILCHIMP_API_KEY)
+
+    # get internal id for the mailing list
+    mailing_list_id = client.list_name_to_unique_id(mailing_list_name)
```

Comment:
I mean, it doesn't matter apart from the fact I have to deal with inefficient bots bringing down our services and so I don't want to ever be thought of as something like that, but can we call this once rather than 4 times (once per option, then again in line 65 before the batch add? :)

Reply:
Moved to a cached function!